### PR TITLE
[deps] bump arrow2 to f07cc2c

### LINKED ISSUE #3746: Bump datafuse-extras/arrow2

### PR DESCRIPTION
**Pull request by PsiACE**
Signed-off-by: Chojan Shang <psiace@outlook.com>

I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

- bump arrow2 step by step
  - [x] f07cc2c remove MutableBuffer, and other updates
  - [x] remove AlignedVec in Databend

## Changelog

- Improvement
- Build/Testing/CI

## Related Issues

Part of #3746 

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Comment by PsiACE** [comment]
The following rev may involve some other changes that affect the review, which I prefer to complete in other PRs.

  - [ ] d14ae86
  - [ ] 9b54146
  - [ ] ?

**Review by sundy-li — APPROVED** [review]

File: query/src/table_functions/numbers_stream.rs [hunk 1]
```diff
@@ -140,16 +140,11 @@ impl NumbersStream {
             None
         } else {
             let size = (current.end - current.begin) as usize;
-            let mut av = AlignedVec::with_capacity(size);
+            let mut av = vec![0; size];
```
Comment (sundy-li) [inline query/src/table_functions/numbers_stream.rs:143]: I think it's ok to use:

```
let av = (current.end - current.begin).collect();

```

File: common/datavalues/src/arrays/primitive/mod.rs [hunk 2]
```diff
@@ -222,16 +222,13 @@ impl<T: DFPrimitiveType> DFPrimitiveArray<T> {
         Self::new(array)
     }
 
-    /// Create a new DataArray by taking ownership of the AlignedVec. This operation is zero copy.
-    pub fn new_from_aligned_vec(values: AlignedVec<T>) -> Self {
+    /// Create a new DataArray by taking ownership of the Vec. This operation is zero copy.
+    pub fn new_from_aligned_vec(values: Vec<T>) -> Self {
```
Comment (sundy-li) [inline common/datavalues/src/arrays/primitive/mod.rs:226]: Better rename to `new_from_vec` ?

File: query/src/table_functions/numbers_stream.rs (hunk 1 above)
Comment (PsiACE) [reply query/src/table_functions/numbers_stream.rs:143]: Let me try

File: common/datavalues/src/arrays/primitive/mod.rs (hunk 2 above)
Comment (PsiACE) [reply common/datavalues/src/arrays/primitive/mod.rs:226]: OK, I'll change it later

**Pull request merged**